lua/telescope/builtin/init.lua

```diff
@@ -1037,4 +1037,48 @@ builtin.current_buffer_tags = function(opts)
   return builtin.tags(vim.tbl_extend("force", {only_current_file = true, hide_filename = true}, opts))
 end
 
+builtin.file_browser = function(opts)
+  opts = opts or {}
+
+  local find_command = opts.find_command
+
+  if not find_command then
+    if 1 == vim.fn.executable("ls") then
+      find_command = { 'ls', '-Fa', '--group-directories-first' }
+    end
+  end
```

Comment:
I think we might even be able to do this with `vim.loop`... perhaps we should find ways to make `vim.loop` handle more of this work. It would be cool to make some of these completely system & command independent as well as not having to start up a new shell or anything like that.

Reply:
Agree, might this shade some light

```lua
local readdir = function(path)
  local files = {}
  local handle = uv.fs_scandir(path)
  if handle == nil then
    return {}
  end

  while true do
    local name, typ = uv.fs_scandir_next(handle)
    if name == nil then
      break
    end
    local is_dir = false
    if typ == 'directory' or typ == 'link' or typ == 'linked' then
      is_dir = true
    end

    local icon, highlight_name = Devicons.get_devicons(name, is_dir)
    icon = (icon and icon ~= '' and icon .. ' ' or '')

    table.insert(files, {
      value = name,
      display = ' ' .. icon .. name .. (is_dir and '/' or ''),
      devicons = {
        icon = icon,
        highlight_name = highlight_name,
      },
      is_dir = is_dir,
    })
  end
  return files
end


```

Reply:
I am currently reading the luv logs, but thanks :smile:. I was just lazy and used `ls` and kinda forgot about `vim.loop` again